services/httpoverrpc/client/client.go

```diff
@@ -0,0 +1,271 @@
+	// Ignore the parent context timeout because we don't want to time out here.
+	ctx = WithoutCancel(ctx)
+	state := args[0].(*util.ExecuteState)
+	if f.NArg() != 1 {
+		fmt.Fprintln(os.Stderr, "Please specify a port to proxy.")
+		return subcommands.ExitUsageError
+	}
+	if len(state.Out) != 1 {
+		fmt.Fprintln(os.Stderr, "Proxying can only be done with exactly one target.")
+		return subcommands.ExitUsageError
+	}
+	port, err := strconv.Atoi(f.Arg(0))
+	if err != nil {
+		fmt.Fprintln(os.Stderr, "Port could not be interpreted as a number.")
+		return subcommands.ExitUsageError
+	}
+
+	proxy := pb.NewHTTPOverRPCClientProxy(state.Conn)
+
+	m := http.NewServeMux()
+	m.HandleFunc("/", func(httpResp http.ResponseWriter, httpReq *http.Request) {
+		var reqHeaders []*pb.Header
+		for k, v := range httpReq.Header {
+			reqHeaders = append(reqHeaders, &pb.Header{Key: k, Values: v})
+		}
+		body, err := io.ReadAll(httpReq.Body)
+		if err != nil {
+			sendError(httpResp, http.StatusBadRequest, err)
+			return
+		}
+		req := &pb.LocalhostHTTPRequest{
+			Request: &pb.HTTPRequest{
+				RequestUri: httpReq.RequestURI,
+				Method:     httpReq.Method,
+				Headers:    reqHeaders,
+				Body:       body,
+			},
+			Port: int32(port),
+		}
+		resp, err := proxy.Localhost(ctx, req)
+		if err != nil {
+			sendError(httpResp, http.StatusInternalServerError, err)
+			return
+		}
+		for _, h := range resp.Headers {
+			for _, v := range h.Values {
+				httpResp.Header().Add(h.Key, v)
+			}
+		}
+		httpResp.WriteHeader(int(resp.StatusCode))
+		if _, err := httpResp.Write(resp.Body); err != nil {
+			fmt.Fprintln(os.Stdout, err)
+		}
+	})
+	l, err := net.Listen("tcp4", p.listenAddr)
+	if err != nil {
+		fmt.Fprintf(state.Err[0], "Unable to listen on %v.\n", p.listenAddr)
+		return subcommands.ExitFailure
+	}
+	fmt.Fprintf(state.Out[0], "Listening on http://%v, ctrl-c to exit...", l.Addr())
+	if err := http.Serve(l, m); err != nil {
+		fmt.Fprintln(os.Stderr, err.Error())
+		return subcommands.ExitUsageError
+	}
+	return subcommands.ExitSuccess
+}
+
+type repeatedString []string
+
+func (i *repeatedString) String() string {
+	if i == nil {
+		return "[]"
+	}
+	return fmt.Sprint([]string(*i))
+}
+
+func (i *repeatedString) Set(value string) error {
+	*i = append(*i, value)
+	return nil
+}
+
+type getCmd struct {
+	method              string
+	headers             repeatedString
+	body                string
+	showResponseHeaders bool
+}
+
+func (*getCmd) Name() string     { return "get" }
+func (*getCmd) Synopsis() string { return "Makes a HTTP call to a port on a remote host" }
+func (*getCmd) Usage() string {
+	return `get [-method METHOD] [-header Header...] [-body body] remoteport request_uri:
+    Make a HTTP request to a specified port on the remote host, defaulting to GET.
+`
+}
+
+func (g *getCmd) SetFlags(f *flag.FlagSet) {
+	f.StringVar(&g.method, "method", "GET", "Address to listen on, defaults to a random localhost port")
```

Comment:
this comment description seems wrong (accidental paste from 'proxy')

Reply:
Thanks, fixed.